### PR TITLE
Use EntityNamePicker in sample templates

### PR DESCRIPTION
**Pull request by mtlewis**
## Hey, I just made a Pull Request!

- Updates the sample templates to use the EntityNamePicker for entity names
- Cleans up some unused config from the sample templates in babcfed.

This PR should not be merged until after the changes in #6723 have been released.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by freben — APPROVED** [review]
Nice! This is ready to merge, isn't it? @mtlewis

**Comment by mtlewis** [comment]
Thinking about it, these sample templates are slated for removal in https://github.com/backstage/backstage/pull/6781 anyway - perhaps we should close this and wait for that to be merged instead?

**Comment by mtlewis** [comment]
Will close for now, feel free to reopen especially if there's some desire to wait longer before merging #6781!

**Pull request closed without merging**
